src/custom-apps/components/WebhookEvents/WebhookEvents.tsx

```diff
@@ -180,59 +187,52 @@ export default WebhookEvents;
 
 type Actions = string[];
 
-export const AsyncWebhookTypes: Record<string, Actions> = {
-  ADDRESS: ["CREATED", "UPDATED", "DELETED"],
-  APP: ["INSTALLED", "UPDATED", "DELETED"],
-  ATTRIBUTE: ["CREATED", "UPDATED", "DELETED"],
-  CATEGORY: ["CREATED", "UPDATED", "DELETED"],
-  CHANNEL: ["CREATED", "UPDATED", "DELETED"],
-  GIFT_CARD: ["CREATED", "UPDATED", "DELETED", "STATUS_CHANGED"],
-
-  CHECKOUT: ["CREATED", "UPDATED", "DELETED"],
-  COLLECTION: ["CREATED", "UPDATED", "DELETED"],
-  CUSTOMER: ["CREATED", "UPDATED", "DELETED"],
-  FULFILLMENT: ["CREATED"],
-  INVOICE: ["DELETED", "REQUESTED", "SENT"],
-  MENU: ["CREATED", "UPDATED", "DELETED"],
-  ORDER: [
-    "CANCELLED",
-    "CONFIRMED",
-    "CREATED",
-    "FULFILLED",
-    "FULLY_PAID",
-    "UPDATED",
-  ],
-  PAGE: ["CREATED", "UPDATED", "DELETED"],
-  PRODUCT: ["CREATED", "UPDATED", "DELETED"],
-  PRODUCT_VARIANT: ["CREATED", "UPDATED", "DELETED"],
-  SALE: ["CREATED", "UPDATED", "DELETED", "TOGGLE"],
-  SHIPPING_PRICE: ["CREATED", "UPDATED", "DELETED"],
-  SHIPPING_ZONE: ["CREATED", "UPDATED", "DELETED"],
-  STAFF: ["CREATED", "UPDATED", "DELETED"],
-  TRANSLATION: ["ACTION_REQUEST", "CREATED", "UPDATED"],
-  VOUCHER: ["CREATED", "UPDATED", "DELETED"],
-  WAREHOUSE: ["CREATED", "UPDATED", "DELETED"],
-};
+const getWebhookTypes = (webhookEvents: string[]) => {
+  const webhookTypes: Record<string, Actions> = {};
+  const multiWords = ["DRAFT_ORDER", "GIFT_CARD", "ANY_EVENTS"];
+
+  const setObject = (
+    keywords: string[],
+    object: string,
+    objectLength: number,
+  ) => {
+    const events = webhookTypes[object] || [];
+    const event = keywords.slice(objectLength).join("_");
+    events.push(!!event.length ? event : object);
+    webhookTypes[object] = events;
+  };
+
+  webhookEvents.map(key => {
+    const keywords = key.split("_");
+
+    // handle 2 word exceptions
+    if (multiWords.includes(keywords.slice(0, 2).join("_"))) {
+      setObject(keywords, keywords.slice(0, 2).join("_"), 2);
```

Comment:
suggestion: `keywords.slice(0, 2).join("_")` is duplicated

Reply:
fixed